engine/src/main/java/org/hibernate/search/bridge/builtin/time/impl/LocalDateNumericBridge.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.bridge.builtin.time.impl;
+
+import java.time.LocalDate;
+
+/**
+ * Converts a {@link LocalDate} to a {@link Long}.
+ * <p>
+ * The digits of the resulting number will have the following pattern: yyyyMMdd
+ * <p>
+ * For example, the date 2001-3-25 will become the number 20010325.
```

Comment:
I'd agree as far as the current Lucene backends are concerned, but even then I find the proposed mapping a bit weird. It's just not the "universal" way for persisting date information. If you are concerned about performance, then we should go for the millis in epoch approach. Personally I'd prefer strings, though.

But it's a different story when taking ES into the picture. There are much higher odds people will access the index directly through the REST API. There we also have to consider inter-op with other apps accessing the index and really should go for a more standard mapping (Dates are ISO-String-based in ES).